tests/files/locations/test_local.py

```diff
@@ -0,0 +1,98 @@
+import os
+import pathlib
+import shutil
+import uuid
+
+import pytest
+
+from astro.constants import FileLocation
+from astro.files.locations import get_location_type, location_factory
+
+CWD = pathlib.Path(__file__).parent
+
+LOCAL_FILEPATH = f"{CWD}/../../data/homes2.csv"
+LOCAL_DIR = f"/tmp/{uuid.uuid4()}/"
+LOCAL_DIR_FILE_1 = str(pathlib.Path(LOCAL_DIR, "file_1.txt"))
+LOCAL_DIR_FILE_2 = str(pathlib.Path(LOCAL_DIR, "file_2.txt"))
+
+sample_filepaths_per_location = [
+    (FileLocation.LOCAL, LOCAL_FILEPATH),
+]
+
+
+@pytest.fixture()
+def local_dir():
+    """create temp dir"""
+    os.mkdir(LOCAL_DIR)
+    open(LOCAL_DIR_FILE_1, "a").close()
+    open(LOCAL_DIR_FILE_2, "a").close()
+    yield
+    shutil.rmtree(LOCAL_DIR)
+
+
+def describe_get_location_type():
+    """test get_location_type()"""
+
+    def with_supported_location():  # skipcq: PTC-W0065, PYL-W0612
+        """With all the supported locations"""
+        location = get_location_type(LOCAL_FILEPATH)
+        assert location == FileLocation.LOCAL
+
+    def with_unsupported_location_raises_exception():  # skipcq: PYL-W0612, PTC-W0065
+        """With all the unsupported locations, should raise a valueError exception"""
+        unsupported_filepath = "invalid://some-file"
+        with pytest.raises(ValueError) as exc_info:
+            _ = get_location_type(unsupported_filepath)
+        expected_msg = "Unsupported scheme 'invalid' from path 'invalid://some-file'"
+        assert exc_info.value.args[0] == expected_msg
+
+
+def describe_is_valid_path():
+    """test is_valid_path() method"""
+
+    def with_supported_filepaths():  # skipcq: PTC-W0065, PYL-W0612
+        """With supported file paths"""
+        location = location_factory(LOCAL_FILEPATH)
+        assert location.is_valid_path(LOCAL_FILEPATH) is True
+
+    def with_unsupported_path_raises_exception():  # skipcq: PYL-W0612, PTC-W0065
+        """With all the unsupported locations, should raise a valueError exception"""
+        nonexistent_file = "/tmp/nonexistent-file"
+        with pytest.raises(ValueError) as exc_info:
+            _ = location_factory(nonexistent_file)
+        expected_msg = "Invalid path: '/tmp/nonexistent-file'"
+        assert exc_info.value.args[0] == expected_msg
```

Comment:
This is testing `BaseLocation.__init__`. We should just `Local.is_valid_path` directly without even using `location_factory`. This can be paramterized under a single `test_isvalid_path` with both positive and negative cases

Reply:
we should have a separate `test_base.py` where we can test `BaseLocation.__init__` for supported an unsupported path

Reply:
Done.